lib/rarity-map.js

```diff
@@ -1,121 +1,113 @@
-module.exports = RarityMap
 
 /**
  * Mapping of torrent pieces to their respective availability in the torrent swarm. Used
  * by the torrent manager for implementing the rarest piece first selection strategy.
  */
-function RarityMap (torrent) {
-  var self = this
+class RarityMap {
+  constructor (torrent) {
+    const self = this
 
-  self._torrent = torrent
-  self._numPieces = torrent.pieces.length
-  self._pieces = []
+    self._torrent = torrent
+    self._numPieces = torrent.pieces.length
+    self._pieces = new Array(self._numPieces).fill(0)
```

Comment:
Any benefits in doing this?

Reply:
You probably don't need the initial `.fill(0)` in `new Array(self._numPieces).fill(0)` maybe just `new Array(self._numPieces)` will do

Reason why i did this was b/c `new Array(n)` is faster then writing to a literal array `[]`<br>And using `.fill(0)` is faster then writing to each index

![skarmavbild 2018-08-26 kl 22 01 14](https://user-images.githubusercontent.com/1148376/44632485-9921c980-a97b-11e8-9e51-1ef390f663b5.png)

also b/c this looks cleaner then
```js
recalculate () {	
  this._pieces.fill(0)
}
```
then
```js
recalculate () {	
  for (let i = 0; i < this._numPieces; ++i) {	
    this._pieces[i] = 0	
  }
}
```